apps/gov-relayer/commands/proposal-pub.ts

```diff
@@ -35,12 +40,22 @@ module.exports = {
 				PROPOSAL_QUEUE
 			);
 
-			monitorNewProposal(cennzApi, mdbClient, (proposalId) => {
-				logger.info("Proposal #%d: Send to queue...", proposalId);
-				proposalQueue.publish(JSON.stringify({ proposalId }), {
-					type: "proposal-new",
+			do {
+				logger.info("Health check: 👌 ok");
+				await monitorNewProposal(cennzApi, mdbClient, (proposalId) => {
+					proposalQueue.publish(JSON.stringify({ proposalId }), {
+						type: "proposal-new",
+					});
+				});
+
+				await monitorProposalActivity(mdbClient, (proposalId) => {
+					proposalQueue.publish(JSON.stringify({ proposalId }), {
+						type: "proposal-activity",
+					});
 				});
-			});
+
+				await waitForBlock(cennzApi, BLOCK_POLLING_INTERVAL);
+			} while (true);
 		} catch (error) {
```

Comment:
Why return to a `while` loop here?

Reply:
the `do...while` is similar to `while`, with one little difference being `do...while` will always execute the block at least once. To be fair, in this scenario, it doesn't really matter but just feels it reads a bit better for me.

Reply:
pushed new changes with `while` loop swaps over to `chain.subscribeFinalizedHeads`